src/helpers.rs

```diff
@@ -827,6 +833,256 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
+                    } else if let (Ok(v), &hir::Ty{
+                        hir_id:_, kind: hir::TyKind::Path(
+                            hir::QPath::Resolved(_, hir::Path { 
+                                span: _, 
+                                res: hir::def::Res::PrimTy(hir::PrimTy::Uint(UintTy::U64)), ..},..)
+                            ), ..
+                    }) = (k.to_u64(), arg_type) {
+                            libffi_args.push((Box::new(v), CArg::UInt64(v)));
+                    } else {
+                        libffi_args.push((Box::new(()), CArg::INVALID));
+                        throw_unsup_format!("Unsupported scalar argument type to external C function: {:?}", cur_arg);
+                    }
+                },
+                _ => {
+                    libffi_args.push((Box::new(()), CArg::INVALID));
+                    throw_unsup_format!("Unsupported argument type to external C function: {:?}", cur_arg);
+                }
+            }
+        }
+
+        // convert them to libffi Arg type
+        let libffi_args = libffi_args.iter()
+            .map(|cur_arg| {
+                match cur_arg {
+                    (boxed, CArg::Int8(_)) => {
+                        arg(&(*boxed.downcast_ref::<i8>().unwrap()))
+                    },
+                    (boxed, CArg::Int16(_)) => {
+                        arg(&(*boxed.downcast_ref::<i16>().unwrap()))
+                    },
+                    (boxed, CArg::Int32(_)) => {
+                        arg(&(*boxed.downcast_ref::<i32>().unwrap()))
+                    },
+                    (boxed, CArg::Int64(_)) => {
+                        arg(&(*boxed.downcast_ref::<i64>().unwrap()))
+                    },
+                    (boxed, CArg::UInt8(_)) => {
+                        arg(&(*boxed.downcast_ref::<u8>().unwrap()))
+                    },
+                    (boxed, CArg::UInt16(_)) => {
+                        arg(&(*boxed.downcast_ref::<u16>().unwrap()))
+                    },
+                    (boxed, CArg::UInt32(_)) => {
+                        arg(&(*boxed.downcast_ref::<u32>().unwrap()))
+                    },
+                    (boxed, CArg::UInt64(_)) => {
+                        arg(&(*boxed.downcast_ref::<u64>().unwrap()))
+                    },
+                    _ => {
+                        // should be unreachable
+                        panic!("Trying to call external function with unsupported type: should have already bailed");
+                    }
+                }
+            }).collect::<Vec<libffi::high::Arg<'_>>>();
+
+        // call the C function using libffi
+        unsafe {
+            let lib = match libloading::Library::new(match this.machine.external_c_so_file.as_ref(){
+                Some(x) => x,
+                None => {
+                    throw_unsup_format!("Error reading specified shared object file");
```

Comment:
This should be "No shared object specified"

Reply:
good catch